sdks/go/pkg/beam/core/runtime/harness/harness.go

```diff
@@ -133,7 +129,17 @@ func Main(ctx context.Context, loggingEndpoint, controlEndpoint string) error {
 			if resp != nil {
 				respc <- resp
 			}
-		}()
+		}
+
+		if req.GetProcessBundle() != nil {
+			// Only process bundles in a gorutine. We need to process plan creating etc
```

Comment:
I'm curious what this need here is. It's certainly simpler to linearize plan creation, but I thought we could handle two plan creations simultaneously, only needing to synchronize on the shared map that holds the plans.